pkg/master/master.go

```diff
@@ -779,6 +785,52 @@ func (m *Master) api_v1() *apiserver.APIGroupVersion {
 	return version
 }
 
+func (m *Master) InstallThirdPartyAPI(rsrc *expapi.ThirdPartyResource) error {
+	kind, group, err := thirdpartyresourcedata.ExtractApiGroupAndKind(rsrc)
+	if err != nil {
+		return err
+	}
+	thirdparty := m.thirdpartyapi(group, kind, rsrc.Versions[0].Name)
+	if err := thirdparty.InstallREST(m.handlerContainer); err != nil {
+		glog.Fatalf("Unable to setup thirdparty api: %v", err)
+	}
+	thirdPartyPrefix := "/thirdparty/" + group + "/"
+	apiserver.AddApiWebService(m.handlerContainer, thirdPartyPrefix, []string{rsrc.Versions[0].Name})
+	thirdPartyRequestInfoResolver := &apiserver.APIRequestInfoResolver{APIPrefixes: util.NewStringSet(strings.TrimPrefix(group, "/")), RestMapper: thirdparty.Mapper}
+	apiserver.InstallServiceErrorHandler(m.handlerContainer, thirdPartyRequestInfoResolver, []string{thirdparty.Version})
+	return nil
+}
+
+func (m *Master) thirdpartyapi(group, kind, version string) *apiserver.APIGroupVersion {
+	resourceStorage := thirdpartyresourcedataetcd.NewREST(m.thirdPartyStorage, group, kind)
+
+	apiRoot := "/thirdparty/" + group + "/"
```

Comment:
@brendandburns, after in-person discussion with @bgrant0607 and @lavalamp, we think thirdparty APIs should also be registered under prefix apis/, and we should stuff the "stable"/"experimental" into the group name. 

For the example in the design doc, the ThirdPartyResource config file should be:

```yaml
metadata:
  name: cron-tab
apiVersion: experimental/v1
kind: ThirdPartyResource
description: "A specification of a Pod to run on a cron style schedule"
versions:
  - name: stable.example.com/v1
  - name: experimental.example.com/v2
```

Then the API server will program in two new RESTful resource paths:
   * `/apis/stable.example.com/v1/namespaces/<namespace>/crontabs/...`
   * `/apis/experimental.example.com/v2/namespaces/<namespace>/crontabs/...`

Do you think it's reasonable?

Sorry for the repeated labor. I can send a PR to patch the design doc if you agree on the new design.